File: android/Detonator/src/main/java/com/iconshot/detonator/helpers/AttributeHelper.java
package com.iconshot.detonator.helpers;

public class AttributeHelper {
    public static Float convertPercentStringToFloat(String percentString) {
        String numberString = percentString.replace("%", "");

        try {
            float value = Float.parseFloat(numberString);

            return value / 100;
        } catch (Exception e) {
            return null;
        }
    }
}

File: ios/Sources/Detonator/Helper/AttributeHelper.swift
public class AttributeHelper {
    public static func convertPercentStringToFloat(_ percentString: String) -> Float? {
        let numberString = percentString.replacingOccurrences(of: "%", with: "")
            .replacingOccurrences(of: "-", with: "")
        
        if let value = Float(numberString) {
            let percent = value / 100
            
            return percentString.starts(with: "-") ? percent * -1 : percent
        } else {
            return nil
        }
    }
}
